### ISSUE TITLE
`unsigned(::Type)` and `signed(::Type)`

### ISSUE THREAD

**Issue by rai-nhdaly** [JuliaLang/julia#30444]
So I've been happily using `unsigned(T)` in generic methods, with the semantics that it will return the unsigned "equivalent" type:
```
julia> unsigned(Int32)
>> UInt32

julia> unsigned(UInt8)
>> UInt8
```

I just found out, though, that A) that's undocumented behavior, and B) there's no `signed(T)` equivalent:
```
julia> signed(UInt8)
ERROR: MethodError: Cannot `convert` an object of type Type{UInt8} to an object of type Signed
```

Apparently the `unsigned(::Type{T})` implementation is coming from here:
https://github.com/JuliaLang/julia/blob/eddcbd7585152c1e404ed0481ffe44c61d504e77/base/multinverses.jl#L9-L14

----

How would we feel about a PR adding the `signed(T)` methods, moving those next to the rest of the `signed`/`unsigned` impls, and adding docs? :slightly_smiling_face:

**Comment by rai-nhdaly** [comment]
> How would we feel about a PR adding the `signed(T)` methods, moving those next to the rest of the `signed`/`unsigned` impls, and adding docs? 🙂

Haha nvm, i just opened #30445! 😄

(Also, sorry for my two github accounts; i still get confused about this... This one is my work account at RelationalAI. 😅)

**Comment by stevengj** [comment]
`signed(T::Type) = typeof(Signed(zero(T)))` seems to work.

**Comment by NHDaly** [comment]
Thanks @stevengj! :)

**Comment by NHDaly** [comment]
Closed in PR #34864 :)

https://github.com/JuliaLang/julia/pull/30445#issuecomment-764613224
